warehouse/query-core/src/main/java/datawave/query/function/KeyToFieldName.java

```diff
@@ -24,43 +23,54 @@ public class KeyToFieldName implements Function<Entry<Key,Value>,Entry<Key,Strin
     
     public KeyToFieldName() {}
     
+    /**
+     * Constructor that sets the includeGroupingContext flag
+     * 
+     * @param includeGroupingContext
+     *            flag that determines if a field name should include the grouping context
+     */
     public KeyToFieldName(boolean includeGroupingContext) {
         this.includeGroupingContext = includeGroupingContext;
     }
     
+    /**
+     * Transforms a key-value entry into a key-string entry
+     * 
+     * @param from
+     *            a key-value entry
+     * @return a key-string entry
+     */
     @Override
     public Entry<Key,String> apply(Entry<Key,Value> from) {
         String fieldName = getFieldName(from.getKey());
-        
         return Maps.immutableEntry(from.getKey(), fieldName);
     }
     
+    /**
+     * Parse the field name from the key's column qualifier
+     * 
+     * @param k
+     *            the key
+     * @return the field name
+     */
     public String getFieldName(Key k) {
         
         int index = -1;
+        byte[] backing = k.getColumnQualifierData().getBackingArray();
         
-        ByteSequence sequence = k.getColumnQualifierData();
-        
-        byte[] arrayReference = sequence.getBackingArray();
-        
-        for (int i = 0; i < sequence.length(); i++) {
-            if (!includeGroupingContext && arrayReference[i] == '.') {
-                index = i;
-                break;
-            }
-            if (arrayReference[i] == 0x00) {
+        for (int i = 0; i < backing.length; i++) {
```

Comment:
The whole point of using the length from the ByteSequence is that the length of the valid bytes in the backing array may not include the entire length of the array.  I think you need to revert this section of code to what it was previously.

Reply:
Reverted in https://github.com/NationalSecurityAgency/datawave/pull/1791/commits/0fa094f4c0ee175dfc8e4c4bf44e0e4877c87a57